Enterprise-CMCS/macpro-mdct-mcr
# Extracting Numerical Field Data

### Description

An lambda for extracting the numerical field data from the MCR & MLR bucket to compare values. First step of a multistep process.

### Related ticket(s)


MDCT-2598

---
### How to test

_Testing locally_
1) You will need to have two consoles open side by side
2) Console 1 - spin up QMR, the console logs here is is how the lambda will communicate back to you
3) Console 2 - Run the two lines below (you may have to modify the first line depending on where you app-api folder is)
> cd documents/github/macpro-mdct-mcr/services/app-api

> aws lambda invoke /dev/null --endpoint-url http://localhost:3002 --function-name app-api-local-numberFieldDataToFile
4) You should see a status 200, the lambda ran successfully.

_Testing on AWS server_
1) Log into cloudtamer
2) On the left hand side menu, select Projects
3) In search box, search for "mdct-mcr-dev". (If this doesn't exist, that's a problem and we need to get you access)
4) There's a lil button with a cloud icon on it and a dropdown arrow, click it.

<img width="300" alt="Screenshot 2023-07-27 at 1 35 11 PM" src="https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/127151429/c39eff48-c694-4fbd-90e4-e64a23e3cad0">

5) From the dropdown, select "Cloud Access Roles"
6) Choose "mdctmcr-application-admin -> Web Access", a new window will popup
7) You are now in AWS
8) In the search box, type "Lambda", select the button in the image below

<img width="646" alt="Screenshot 2023-07-27 at 1 38 10 PM" src="https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/127151429/0a14629d-def5-4d9d-a9fc-853eb5c52a9b">

9) In the search filter, input "app-api-mdct-2598-strip-num-numberFieldDataToFile" and select the result
10) Press the "test" in the red box first and then the "test" in the green box will be avaliable.

<img width="1366" alt="Screenshot 2023-07-27 at 1 42 31 PM" src="https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/127151429/73a10eef-dbec-4eb0-9de4-847ed13862f8">

11) The lambda will run and it'll take about 22 seconds give or take
12) Once it's finished a new file will be generated. 
13) To see the file, go to "S3" -> filter to "database-mdct-2598-strip-num-mlr" -> "fieldData/" -> "numberValues.json"


### Important updates


N/A

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
